### PR TITLE
GOPAY-3782: Add support for transaction cancellation

### PR DESCRIPTION
**Pull request by taugusto-godaddy**
Add cancel transaction support

### CODE REVIEW

File: lib/transactions.js [hunk 1]
```diff
@@ -118,6 +118,46 @@ module.exports.getTransactionCurrency = function getTransactionCurrency(options,
...
+        cancelRequest: options.cancelRequest
+      },
+      requestId: options.requestId
+    },
+    function (err, transaction) {
+      if (err) {
+        return next(err);
+      }
+
+      if (!transaction || transaction.status !== "DECLINED") {
```
Comment (isiddarta-godaddy) [inline lib/transactions.js:153]:
```suggestion
      if (!transaction || transaction.status !== "VOIDED" || transaction.status !== "REFUNDED" ) {
```

Comment (isiddarta-godaddy) [reply lib/transactions.js:153]: or maybe we can check if transaction is still `AUTHORIZED` or `CAPTURED` , then we throw `Cancel failed`. Im not 100% sure which one cover majority of cases

Comment (taugusto-godaddy) [reply lib/transactions.js:153]: Weird... in my tests the transaction status went as DECLINED 🤔

**Review by isiddarta-godaddy — APPROVED** [review]

File: lib/transactions.js (hunk 1 above)
Comment (isiddarta-godaddy) [reply lib/transactions.js]: cancel API has some different cases where it returns specific status

**Pull request merged**